lib/streamlit/connections/snowpark_connection.py

```diff
@@ -79,11 +79,21 @@ def __init__(self, connection_name: str, **kwargs) -> None:
         with self._lock:
             super().__init__(connection_name, **kwargs)
 
-    # TODO(vdonato): Teach the .connect() method how to automagically connect in a SiS
-    # runtime environment.
     def _connect(self, **kwargs) -> "Session":
+        from snowflake.snowpark.context import get_active_session  # type: ignore
+        from snowflake.snowpark.exceptions import (  # type: ignore
+            SnowparkSessionException,
+        )
         from snowflake.snowpark.session import Session
 
+        # If we're in a runtime environment where there's already an active session
+        # available, we just use that one. Otherwise, we fall back to attempting to
+        # create a new one from whatever credentials we have available.
+        try:
+            return get_active_session()
```

Comment:
Is there a situation where the session returned by `get_active_session` is actually not what the user is trying to connect to? E.g. if the user is logged in with a certain account/role, but actually wants to use a different one?

Reply:
Not as far as I understand -- @sfc-gh-jcarroll might know if there's any weirdness that can go on here, though

Reply:
I'm not aware of any either, I think this will work. Might be worth tagging Yijun on the main PR to look at the Snowpark connection all up for any concerns like this?